squad/frontend/templates/squad/failures.jinja2

```diff
@@ -23,26 +27,27 @@
             <th>{{ e.slug }}</th>
             {% endfor %}
         </thead>
-        {% if not ufailures %}
+        {% if fc.failures()|length == 0 %}
```

Comment:
`fc.failures()` is the heaviest query and you are already calling it on line 21 in squad/frontend/failures.py

Reply:
I changed this to `fc.failures().exists()` instead. Thanks for catching this.

Reply:
Does `.exists()` re-use the same query? I think it'll still re-run the query + `LIMIT 1`